garcon/activity.py

```diff
@@ -253,21 +255,37 @@ def run(self):
         previous activity is consumed (context).
         """
 
-        try:
-            activity_task = self.poll()
-        except Exception as error:
+        max_try = 5
```

Comment:
This feels like an arbitrary number, so it would be better to have this being a configuration that you can set and provide a constant that holds a default value.